gk/fib.c

```diff
@@ -1664,3 +1665,303 @@ del_fib_entry(const char *ip_prefix, struct gk_config *gk_conf)
 
 	return del_fib_entry_numerical(&prefix_info, gk_conf);
 }
+
+struct gk_fib_ether_entry {
+
+	bool          stale;
+
+	/* The IP address of the nexthop. */
+	struct ipaddr nexthop_ip;
+
+	uint32_t      ref_cnt;
+
+	/* The whole link-layer header. */
+	struct {
+		/* Ethernet header (required). */
+		struct ether_hdr eth_hdr;
+		/* VLAN header (optional). */
+		struct vlan_hdr  vlan_hdr;
+	} __attribute__((packed)) l2_hdr;
```

Comment:
The field `l2_hdr` is an implementation aspect of the current FIB, so we shouldn't export it to Lua since it may go away. We only need to export the MAC address of `nexthop_ip`.